### ISSUE TITLE
Validate Gateway API resources for Kuma support

### ISSUE THREAD

**Issue by michaelbeaumont** [kumahq/kuma#3464]
### Description

We shouldn't do any validation inside our controllers that we can instead do in a separate admission webhook.

**Comment by jpeach** [comment]
Yup. The validation can use the upstream package https://github.com/kubernetes-sigs/gateway-api/tree/master/apis/v1alpha2/validation.

**Comment by michaelbeaumont** [comment]
@lahabana No, the upstream webhook in #3416 would validate the resources for the Gateway API spec and this issue would be to _further_ validate those resources for the Kuma requirements.

**Comment by michaelbeaumont** [comment]
Note that a lot of validation is done already in the controllers.

**Comment by michaelbeaumont** [comment]
I'm going to close this for now, validation can be added later and our controllers already validate resources.
